### PR TITLE
Helm release

### PR DESCRIPTION
**Pull request by aaron-prindle**
this is a PR outlining the process and changes required.  This should not be merged until the PR for  https://github.com/grafeas/kritis/issues/136 is merged.

### CODE REVIEW

File: deploy/cloudbuild-release.yaml [hunk 1]
```diff
@@ -3,4 +3,20 @@ steps:
...
            "-t", "gcr.io/kritis-project/kritis-server:$TAG_NAME", "."]
-images: ["gcr.io/kritis-project/kritis-server:$TAG_NAME"]
+  - name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-f", "helm-hooks/Dockerfile",
+           "-t", "gcr.io/kritis-project/preinstall:$TAG_NAME", ".",
+           "--build-arg", "stage=preinstall"]
+  - name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-f", "helm-hooks/Dockerfile",
+           "-t", "gcr.io/kritis-project/postinstall:$TAG_NAME", ".",
+           "--build-arg", "stage=preinstall"]
```
Comment (priyawadhwa) [inline deploy/cloudbuild-release.yaml]: We should probably add the predelete image here as well!

File: deploy/cloudbuild.yaml [hunk 2]
```diff
@@ -8,10 +8,16 @@ steps:
...
            "-t", "gcr.io/kritis-project/preinstall:${COMMIT_SHA}", "."]
   - name: "gcr.io/cloud-builders/docker"
     args: ["tag", "gcr.io/kritis-project/preinstall:${COMMIT_SHA}", "gcr.io/kritis-project/preinstall:latest"]
+  # Build postinstall container
+  - name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-f", "helm-hook/postinstall/Dockerfile",
+           "-t", "gcr.io/kritis-project/postinstall:${COMMIT_SHA}", "."]
+  - name: "gcr.io/cloud-builders/docker"
+    args: ["tag", "gcr.io/kritis-project/postinstall:${COMMIT_SHA}", "gcr.io/kritis-project/postinstall:latest"]
   # Build container to build resolve-tags
```
Comment (priyawadhwa) [inline deploy/cloudbuild.yaml]: Same as above, we should release the predelete image as well :)

File: deploy/cloudbuild-release.yaml (hunk 1 above)
Comment (aaron-prindle) [reply deploy/cloudbuild-release.yaml]: done

File: deploy/cloudbuild.yaml (hunk 2 above)
Comment (aaron-prindle) [reply deploy/cloudbuild.yaml]: done

**Comment by aaron-prindle** [comment]
This PR adds helm releases to the cloudbuild-release.yaml which has a corresponding trigger on git tags pushed upstream.  The tag regex is kritis-vX.Y.Z.  This PR also modifies the Makefile to use have a 'build-and-push-image' rule.  The default tag for images from the Makefile is no longer :latest but the COMMIT_SHA256 to ensure that the images used are from the branch HEAD, also to account for possible race conditions when two PRs are added simultaneously and :latest could be overridden.  Using the Makefile now to build locally and 'helm install ./kritis-charts' as you were doing previously you should run:
```
IMAGE_TAG=v0.0.1 REGISTRY=gcr.io/MY_PROJECT build-and-push-image
helm install ./kritis-charts
```

**Pull request merged**